gulpfile.js

```diff
@@ -297,6 +302,218 @@ gulp.task('minifyRelease', ['generateStubs'], function() {
     });
 });
 
+gulp.task('deploy', function(done) {
+    var cacheControl = "max-age=3600";
+    var concurrencyLimit = 2000;
+
+    var argv = yargs.usage('Usage: delpoy -b [Bucket Name] -d [Upload Directory]')
+        .demand(['b', 'd']).argv;
+
+    var uploadDir = argv.d;
+    var bucketName = argv.b;
+
+    var s3 = new aws.S3({
+        maxRetries: 10,
+        retryDelayOptions: {
+            base: 500
+        }
+    });
+
+    var existingBlobs = [];
+    var totalFiles = 0;
+    var uploaded = 0;
+
+    // get all files currently in bucket
+    function listAll(callback, marker) {
+        return s3.listObjects({
+            Bucket : bucketName,
+            MaxKeys : 1000,
+            Prefix: uploadDir,
+            Marker : marker
+        }, function(err, data) {
+            if (err) {
+                console.log(err);
+            } else {
+                for (var i = 0; i < data.Contents.length; i++) {
+                    existingBlobs.push(data.Contents[i].Key);
+                }
+                if (data.IsTruncated) {
+                    // get next page of results
+                    return listAll(callback, existingBlobs[existingBlobs.length - 1]);
+                } else {
+                    return callback();
+                }
+            }
+        });
+    }
+
+    // delete unneeded files in bucket
+    function clean() {
+        return new Promise(function(resolve, reject){
+            console.log('Cleaning up old files...');
+            if (existingBlobs.length > 0) {
+                var objects = [];
+                for (var i = 0; i < existingBlobs.length; i++) {
+                    var blob = existingBlobs[i];
+                    objects.push({Key: blob});
+                }
+
+                var params = {
+                    Bucket: bucketName,
+                    Delete: {
+                        Objects: objects
+                    }
+                };
+
+                s3.deleteObjects(params, function(err, data) {
+                    if (err) {
+                        console.log(err);
+                        return reject(err);
+                    } else {
+                        resolve();
+                    }
+                });
+            }
+            resolve();
+        });
+    }
+
+    return listAll(function() {
+        var files = globby.sync([
+            '**',
+            '!node_modules/**',
+            '!Tools/**',
+            '!.git*',
+            '!.git/**'
+        ], {
+            dot: true // include hidden files
+        });
+
+        return async.forEachLimit (files, concurrencyLimit, function(filename, callback) {
+            var blobName = uploadDir + '/' + filename;
+            var contentType = mime.lookup(filename);
+            var compress = compressible(contentType);
```

Comment:
I don't think either the `mime` or `compressible` modules have the same behavior as the simple class that we had in the old tool.  For example, does this correctly detect terrain, geojson and czml?  Assuming it's just using the extension, I highly doubt it.  While I'm all for using third-party libraries, since we only have a few types to worry about it might be easier to just port the old behavior directly.